### ISSUE TITLE
Mounting an engine before {{outlet}} causes error

### ISSUE THREAD

**Issue by alecho** [dgeb/ember-engines#68]
In `app/templates/application.hbs`:
```
<h2 id="title">Welcome to Ember</h2>
{{#link-to "index"}}Home{{/link-to}}
{{#link-to "boom"}}Click-click{{/link-to}}   <-- link to routable engine
{{mount "mountable-engine"}}
{{outlet}}
```

causes `TypeError: undefined is not an object (evaluating 'node.getState().controller.set')`

while everything seems fine if this is the case:
```
<h2 id="title">Welcome to Ember</h2>
{{#link-to "index"}}Home{{/link-to}}
{{#link-to "boom"}}Click-click{{/link-to}}
{{outlet}}
{{mount "mountable-engine"}}
```

**Comment by alecho** [comment]
Example repo: https://github.com/alecho/ember-engine-poc Commit 853aa6605ad7057f283e567893d0107032087e40 is working but switching the code to the first example above demonstrates the issue.

**Comment by knownasilya** [comment]
Sounds like it's related to https://github.com/dgeb/ember-engines/pull/66

**Comment by dgeb** [comment]
Thanks for reporting. This _should_ be fixed in v0.1.3 (via #66), but please let us know if you experience other problems.
